tests/test_tui_package_contract.py

```
@@ -0,0 +1,262 @@
+    assert result.returncode == 0, result.stderr
+
+
+def test_host_npm_target_rejects_unknown_architecture() -> None:
+    for system in ("Linux", "Darwin"):
+        with mock.patch.object(platform, "system", return_value=system), mock.patch.object(
+            platform, "machine", return_value="riscv64"
+        ):
+            try:
+                host_npm_target()
+            except RuntimeError as error:
+                assert str(error) == f"unsupported test host: {system.lower()}-riscv64"
```

Ruff (0.16.1)

[warning] Found assertion on exception `error` in `except` block, use `pytest.raises()` instead

(PT017)